composition-js/src/merging/merge.ts

```diff
@@ -358,10 +353,19 @@ class Merger {
     // If we get here with `nameInSupergraph` unset, it means there is no usage for the directive at all and we
     // don't bother adding the spec to the supergraph.
     if (nameInSupergraph) {
+      const specInSupergraph = compositionSpec.supergraphSpecification();
       const errors = linkSpec.applyFeatureToSchema(this.merged, specInSupergraph, nameInSupergraph === specInSupergraph.url.name ? undefined : nameInSupergraph, specInSupergraph.defaultCorePurpose);
       assert(errors.length === 0, "We shouldn't have errors adding the join spec to the (still empty) supergraph schema");
+      const argumentsMerger = compositionSpec.argumentsMerger?.call(this, this.merged);
```

Comment:
I believe that the first argument should be null here since `argumentsMerger` is not part of an object.

```suggestion
      const argumentsMerger = compositionSpec.argumentsMerger?.call(null, this.merged);
```

Reply:
I believe you're right, thanks.